customlists/customlist_export.py

```diff
@@ -0,0 +1,484 @@
+import argparse
+import json
+import logging
+import os
+import re
+from typing import IO, Iterable, List, Optional, Union
+
+import feedparser
+import jsonschema
+import requests
+from requests import Session
+
+
+class CollectionReference:
+    _id: int
+    _id_updated: int
+    _name: str
+    _protocol: str
+
+    def __init__(self, id: int, name: str, protocol: str):
+        self._id = id
+        self._id_updated = id
+        self._name = name
+        self._protocol = protocol
+
+    def to_dict(self) -> dict:
+        return {
+            "%type": "collection",
+            "id": self._id,
+            "name": self._name,
+            "protocol": self._protocol,
+        }
+
+    def id_original(self) -> int:
+        return self._id
+
+    def id(self) -> int:
+        return self._id_updated
+
+    def name(self) -> str:
+        return self._name
+
+    def protocol(self) -> str:
+        return self._protocol
+
+    def update_id(self, id: int):
+        assert type(id) == int
```

Comment:
To be clear, these asserts were added because they caught silly mistakes made during development. They're not intended to perform validation, so it's fine if they vanish during compilation.